src/main/java/org/opensearch/neuralsearch/processor/normalization/ZScoreNormalizationTechnique.java

```diff
@@ -52,12 +72,36 @@ public void normalize(NormalizeScoresDTO normalizeScoresDTO) {
             for (int j = 0; j < topDocsPerSubQuery.size(); j++) {
                 TopDocs subQueryTopDoc = topDocsPerSubQuery.get(j);
                 for (ScoreDoc scoreDoc : subQueryTopDoc.scoreDocs) {
-                    scoreDoc.score = normalizeSingleScore(scoreDoc.score, zscores.stdPerSubquery[j], zscores.meanPerSubQuery[j]);
+                    scoreDoc.score = normalizeSingleScore(
+                        scoreDoc.score,
+                        zscores.stdPerSubquery[j],
+                        zscores.meanPerSubQuery[j],
+                        zscores.maxPerSubQuery[j],
+                        zscores.minPerSubQuery[j]
+                    );
                 }
             }
         }
     }
 
+    @Override
+    public void validateCombinationTechnique(ScoreCombinationTechnique combinationTechnique) throws IllegalArgumentException {
```

Comment:
this approach is good, but I think we can do it more elegantly. How about validating it as part of the norm technique constructor? We can switch from Function to BiFunction, or create a new dto class  and leave Function, both approaches are fine. Then we pass combination technique to norm technique constructor in a factory. This way you can get rid of the validate method, as it doesn't do match for all techniques except the z_score

Reply:
Used the DTO pattern